### PR TITLE
Subtotals and headings now have a position argument

### PR DESCRIPTION
**Pull request by jonkeane**
The position argument takes top/bottom/relative as values. If relative (the default) is given, then an after argument must be present.

### CODE REVIEW

File: R/insertions.R [hunk 1]
```diff
@@ -205,7 +214,7 @@ setMethod("func", "Heading", function (x) return(NA))
 #' @rdname Insertions
 #' @export
 setMethod("anchors", "Insertions", function (x) {
-    f <- vapply(x, anchor, integer(1))
+    f <- vapply(x, anchor, 1)
```
Comment (nealrichardson) [inline R/insertions.R]: This line looks odd.

File: R/subtotals-and-headings.R [hunk 2]
```diff
@@ -91,74 +95,89 @@
...
+    return(new("Subtotal", list(name = name,
+                                categories = categories,
+                                position = position,
+                                after = after)))
+}
+
+validatePosition <- function (position, after) {
+    if (position == "relative") {
+        # if position is realtive, make sure there is an after
+        if(is.null(after)) {
```
Comment (nealrichardson) [inline R/subtotals-and-headings.R]: Space between `if` and `(`

File: R/subtotals-and-headings.R [hunk 3]
```diff
@@ -91,74 +95,89 @@
...
-                      ". Did you mean to make a Subtotal?")
-    } else {
-        val <- TRUE
-    }
+#' @rdname SubtotalsHeadings
+#' @export
+Heading <- function (name,
+                     position = c("relative", "top", "bottom"),
+                     after = NULL) {
+    # match.args position
```
Comment (nealrichardson) [inline R/subtotals-and-headings.R]: I'm not sure this comment adds any value

File: R/insertions.R (hunk 1 above)
Comment (jonkeane) [reply R/insertions.R]: Oph, yeah. Since `anchor()` could now return either an integer or a character I've switched this to `unlist(lapply())`

**Pull request merged**